workspaces/tests/deploy_project.rs

```diff
@@ -0,0 +1,27 @@
+#![cfg(feature = "unstable")]
+#![recursion_limit = "256"]
+use test_log::test;
+use workspaces::prelude::*;
+
+#[test(tokio::test)]
+async fn test_dev_deploy_project() -> anyhow::Result<()> {
+    let worker = workspaces::sandbox();
+    let wasm = workspaces::compile_project("./tests/test-contracts/status-message").await?;
```

Comment:
TBH, this path is a bit confusing.. it's not immediately clear that it's relative to the `workspaces` folder (it reads as though this path points to `workspaces/tests/tests/test-contracts/status-message`).

Ideally, given the location of this file, shouldn't the contract path be `./test-contracts/status-message`?

I'm thinking maybe we should wrap the `compile_project` function in a macro like `compile!()` that internally checks `file!()`, getting the parent and resolving relative paths from there (leaving paths as-is if they're already absolute).

Something like this:

```rust
#[macro_export]
macro_rules! compile {
    ($contract_path:expr) => {
        $crate::__compile_contract(::std::path::Path::new(file!()), $contract_path)
    };
}
```

Although there's an issue when using a cargo workspace with members (as with the case of `workspaces-rs`), because `file!()` returns the path relative to the root workspace and not the member crate itself. So, from `deploy_project.rs`, `parent(file!())` returns `workspaces/tests`

I'm thinking we could use `cargo-metadata` to first, return all the members, with their respective paths, we can then find the member that owns the path returned by `parent(file!())` and resolve the relative path from there.

Although, you'd have to introduce new dependencies like [`pathdiff`](https://docs.rs/pathdiff/0.2.1/pathdiff/fn.diff_paths.html) to resolve relative paths, seeing as the stdlib doesn't have that functionality.

Reply:
Resolved in https://github.com/near/workspaces-rs/pull/89